File: mock-facility-management/src/main/java/de/vkb/dojo/es/facilityManagement/model/feedback/UnknownFeedback.java
package de.vkb.dojo.es.facilityManagement.model.feedback;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonTypeName;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("unknown")
public class UnknownFeedback implements Feedback {
    @JsonCreator
    public UnknownFeedback() {
    }

    @JsonGetter
    public String getMessage() {
        return "retry in a few seconds";
    }

    public Boolean getSuccess() {
        return false;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/common/model/feedback/UnknownFeedback.kt
package de.vkb.dojo.es.common.model.feedback

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("unknown")
class UnknownFeedback: Feedback {
    override val success = false
    val message = "retry in a few seconds"
}

